cityofaustin/atd-moped
# Establish baseline components code changes (#1)

### Note: 🚧  This PR is now being used to collect new features before releasing the updated moped components UI and database updates. This should only be merged once we test with the transform script for existing components and their features. 🚧 

Pre-merge steps:
- [x] ~Fix `Line 108:6:  React Hook useEffect has a missing dependency: 'errorMessageDispatch'. Either include it or remove the dependency array. If 'errorMessageDispatch' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps` build error. I will do this in one of the incoming PRs.~ Fixed in https://github.com/cityofaustin/atd-moped/pull/933
- [x] Test with output of the transform script
- [x] Pull main and resolve conflicts (make sure migrations don't break locally)
- [x] ~Update DB migration timestamps to avoid any conflicts there (should be safe since nothing else has changed this but we should be careful about this)~ Test applying migrations to local DB populated with staging and production snapshots

Post-merge steps:
- [ ] Run transform script on staging DB

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/10400

This PR moves John's base for the project components updates to the project view folders. I kept the old code around since it will be needed to migrate existing components.

@johnclary - the only major changes to your code are:
- Wrapped `MapView` in a MUI full screen Dialog component
- Passing project name and project statuses props down from the `ProjectView` component for the map status badge (project ID comes from the URL)
- Splitting the `settings.js` into separate map settings and map style settings (we'll see how this works since they rely on a lot of common variables - we can always bring them back together)
- I couldn't get the "Close Map" button to go back to where the user came from and also fall back to the Summary tab - so it links back to the Summary tab every time for now. I can create an issue to come back to this, but it boils down how we use the `useRoutes` RR hook and me not finding a great way to inject the last route visited into the navigation state (if there is a last route -> go there, if not -> go to Summary tab). Found [this possible workaround](https://stackoverflow.com/a/71647428).

## Testing
**URL to test:** 

https://md-10400-baseline-components--atd-moped-main.netlify.app/moped/ (or test locally)

**Steps to test:**
1. Go to an existing project or create a new one
2. Go to the **Map** tab and a full screen map view should open
3. Check that the project name and project # match the project the map linked from
4. Click the **Close Map X** button in the top right corner and you should navigate back to the **Summary** view

---
#### Ship list
